django/views/generic/list.py

```diff
@@ -181,7 +181,13 @@ def get_template_names(self):
         if hasattr(self.object_list, 'model'):
             opts = self.object_list.model._meta
             names.append("%s/%s%s.html" % (opts.app_label, opts.model_name, self.template_name_suffix))
-
+        elif not isinstance(self.object_list, QuerySet) and not names:
+                raise ImproperlyConfigured(
```

Comment:
the indentation on this line is incorrect - I'm not sure why pep8 didn't catch this :(